policy/pa.go

```diff
@@ -434,12 +441,16 @@ func (pa *AuthorityImpl) WillingToIssue(domains []string) error {
 //
 // If multiple domains are invalid, the error will contain suberrors specific to
 // each domain.
-func WellFormedDomainNames(domains []string) error {
+func WellFormedIdentifiers(idents []identifier.ACMEIdentifier) error {
 	var subErrors []berrors.SubBoulderError
-	for _, domain := range domains {
-		err := ValidDomain(domain)
-		if err != nil {
-			subErrors = append(subErrors, subError(domain, err))
+	for _, ident := range idents {
+		if ident.Type == identifier.TypeDNS {
+			err := ValidDomain(ident.Value)
+			if err != nil {
+				subErrors = append(subErrors, subError(ident, err))
+			}
+		} else {
+			subErrors = append(subErrors, subError(ident, errUnsupportedIdentifier))
```

Comment:
for the future: when this gets a third case for TypeIP, this will be more elegant as a switch/case rather than if/elseif/else.